Source/WebInspectorUI/UserInterface/Views/FontDetailsPanel.js

```diff
@@ -538,6 +552,28 @@ WI.FontDetailsPanel = class FontDetailsPanel extends WI.StyleDetailsPanel
         ], WI.UIString("Normal", "Normal @ Font Details Sidebar Property Value", "Property value for any `normal` CSS value."));
     }
 
+    _listAllWebpageFonts() {
+        let allElements = document.querySelectorAll("*");
+        let fontSet = new Set();
+
+        allElements.forEach(element => {
+            let computedStyle = window.getComputedStyle(element);
+
+            let fontFamily = computedStyle.getPropertyValue("font-family");
+
+            let fontNames = fontFamily.split(",").map(name => name.trim());
+
+            fontNames.forEach(name => {
+                fontSet.add(name);
+            });
+        });
+
+        let fontList = Array.from(fontSet);
+        fontList.sort();
+
+        return fontList;
+    }
+
```

Comment:
This won't do what you think it does. 

The scope of this code is Web Inspector's `Main.html` page, not the inspected page. So, at best, this lists all the strings defined in the `font-family` CSS property of any CSS rule in the Web Inspector tool itself.

To solve this sort of task, a client-side solution is insufficient. 

As mentioned by @dcrousso in Slack, there needs to be a way on the backend (C++) to aggregate the fonts actually loaded and used. Simply iterating through `font-family` string values doesn't guarantee that those fonts 1) were loaded or 2) were used. For example, `font-family: "Nothing"` doesn't say that the font file itself was actually loaded as part of a `@font-face` CSS rule.

The task at hand involves a bit of digging through the backend and identifying where that data exists (@litherum might be able to point you in the right direction). Then, collect that data about used fonts and keep it updated when something on the page changes, like a styles recalculation. Then, the necessary `*Agent`, [InspectorInstrumentation.cpp](https://github.com/WebKit/WebKit/blob/main/Source/WebCore/inspector/InspectorInstrumentation.cpp) and protocol files need to be updated to transport that data from the backend to the client (perhaps [CSS.json](https://github.com/WebKit/WebKit/blob/main/Source/JavaScriptCore/inspector/protocol/CSS.json#L339-L349)) . Then, ultimately  render the list of fonts in the Font sidebar panel.